### PR TITLE
Update/25.0.4

### LINKED ISSUE #21: [feat] Set Up `cron` Jobs
Warnings about background tasks are annoying and confusing.  `cron` must be set up on the backend and deployed as an update to eOS Nextcloud package. 

https://docs.nextcloud.com/server/25/admin_manual/configuration_server/background_jobs_configuration.html#cron

This is a blocker for using the News (RSS Feed) Nextcloud App

### LINKED ISSUE #27: [feat] Enable Backend Features for Memories App
https://vuejsexamples.com/yet-another-photo-management-app-for-nextcloud/

### PR DESCRIPTION
**Pull request by kn0wmad**
## Server update
- [x] Update server to v25.0.4
## Memories (Photo app)
- [x] Up PHP memory limit to 3GB
- [x] Add support for more [file types](https://github.com/pulsejet/memories/wiki/File-Type-Support)
- [x] [Add deps and config](https://github.com/pulsejet/memories/wiki/Configuration) for Memories
- [x] Create [Indexing Action for Memories](https://github.com/pulsejet/memories#-installation)
- [x] Create Action to download machine learning models for [Recognize App](https://github.com/nextcloud/recognize)
- [x] Create Action to setup reverse geotagging for Map within Memories application
- [x] Add WebDAV URLs to Properties
- [x] Fix `cron` bug
- [x] Add 'Log Level' option to Config
- [x] New migration

Closes #21 
Closes #27

### CODE REVIEW

File: actions/index-memories.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    \"version\": \"0\",
+    \"message\": \"Success!  Photos have been indexed and video transcoding and HLS are now enabled for the Memories application.  Please restart your Nextcloud service for changes to take effect. \",
+    \"value\": null,
+    \"copyable\": false,
+    \"qr\": false
+}"
+
+# Run the occ command to index photos for the Memories app
+sudo -u www-data php $NEXTCLOUD_DIR/occ memories:index -f > /dev/null 2>&1
+yes | sudo -u www-data php $NEXTCLOUD_DIR/occ memories:video-setup > /dev/null 2>&1
```
Comment (chrisguida) [inline actions/index-memories.sh:18]: love it

File: actions/places-setup.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+action_result_running="    {
+    \"version\": \"0\",
+    \"message\": \"Success!  You can now use the Map inside your Memories application. \",
+    \"value\": null,
+    \"copyable\": false,
+    \"qr\": false
+}"
+
+# Run the occ command to setup map for the Memories app
+yes | sudo -u www-data php /var/www/html/occ memories:places-setup > /dev/null 2>&1
```
Comment (chrisguida) [inline actions/places-setup.sh:17]: this one is also required to be interactive?

File: scripts/services/migrations.ts [hunk 3]
```diff
@@ -1,10 +1,9 @@
+// deno-lint-ignore-file
```
Comment (chrisguida) [inline scripts/services/migrations.ts:1]: why?

File: scripts/services/migrations.ts [hunk 4]
```diff
@@ -31,6 +30,14 @@ export const migration: T.ExpectedExports.migration = compat.migrations
...
+          false,
+          { version: "25.0.4", type: "up" },
+        ),
+        down: () => { throw new Error('Downgrades prohibited') },
+      },
     },
-    "25.0.3.3",
-  );
+    "25.0.4",
+  );
```
Comment (chrisguida) [inline scripts/services/migrations.ts:43]:
```suggestion
  );

```

File: scripts/services/properties.ts [hunk 5]
```diff
@@ -1,2 +1,3 @@
-import { compat } from "../deps.ts";
-export const properties = compat.properties
\ No newline at end of file
+import { compat, types as T } from "../deps.ts";
+
+export const properties: T.ExpectedExports.properties = compat.properties
```
Comment (chrisguida) [inline scripts/services/properties.ts:3]:
```suggestion
export const properties: T.ExpectedExports.properties = compat.properties

```

**Review by chrisguida — APPROVED** [review]
Some minor nits, questions, and suggestions. Otherwise looks great.

File: actions/places-setup.sh (hunk 2 above)
Comment (kn0wmad) [reply actions/places-setup.sh:17]: Yes, iirc, it is asking if you're okay with the size of the download - which is expressed in the Action warning

File: scripts/services/properties.ts (hunk 5 above)
Comment (kn0wmad) [reply scripts/services/properties.ts:3]: This looks identical - what am I missing?

File: docker_entrypoint.sh [hunk 6]
```diff
@@ -119,6 +156,8 @@ if [ -e "$FILE" ] ; then {
 } 
 fi
 
+exec busybox crond -f -l 0 -L /dev/stdout
```
Comment (chrisguida) [inline docker_entrypoint.sh:182]: I would guess this `exec` breaks sigterm handling....

File: scripts/services/properties.ts (hunk 5 above)
Comment (chrisguida) [reply scripts/services/properties.ts:3]: Missing newline, see the red circle with the red horizontal line, that means the newline is missing

**Review by chrisguida — CHANGES_REQUESTED** [review]

File: scripts/services/migrations.ts (hunk 3 above)
Comment (kn0wmad) [reply scripts/services/migrations.ts:1]: @MattDHill do we want this?  I think your vscode added it for some reason

Comment (MattDHill) [reply scripts/services/migrations.ts:1]: Can delete

File: docker_entrypoint.sh (hunk 6 above)
Comment (kn0wmad) [reply docker_entrypoint.sh]: Fixed

File: docker_entrypoint.sh [hunk 7]
```diff
@@ -108,8 +175,9 @@ if [ -e "$FILE" ] ; then {
   chmod -R 0600 /var/lib/postgresql/.pgpass
   # Installing Nextcloud Frontend
   echo "Configuring frontend..."
-  sed -i '/echo "Initializing finished"/a touch re.start && echo "Follow the White Rabbitcoin." > \/re.start' /entrypoint.sh 
+  sed -i '/echo "Initializing finished"/a touch re.start && echo "Follow the White Rabbit." > \/re.start' /entrypoint.sh
```
Comment (chrisguida) [inline docker_entrypoint.sh:178]: lol

File: manifest.yaml [hunk 8]
```diff
@@ -134,5 +140,45 @@ actions:
...
+    allowed-statuses:
+      - running
+    implementation:
+      type: docker
+      image: main
+      system: false
+      entrypoint: download-models.sh
+      inject: true
+      args: []
+      io-format: json
```
Comment (chrisguida) [inline manifest.yaml:184]:
```suggestion
      io-format: json

```

**Review by chrisguida — APPROVED** [review]
Ship it

File: manifest.yaml (hunk 8 above)
Comment (chrisguida) [reply manifest.yaml:184]: Another end-of-file newline

**Pull request merged**